src/client/v2/algod/accountApplicationInformation.ts

```diff
@@ -24,6 +24,8 @@ export default class AccountApplicationInformation extends JSONRequest<
 
   // eslint-disable-next-line class-methods-use-this
   prepare(body: Record<string, any>): AccountApplicationResponse {
-    return AccountApplicationResponse.from_obj_for_encoding(body);
+    return AccountApplicationResponse.fromEncodingData(
```

Comment:
Yes this is pretty verbose and not very nice.

When decoding from a serialized JSON string, [`decodeJSON`](https://github.com/algorand/js-algorand-sdk/blob/f79528e68ce60e0a1142fab4c67e6ebdb4379caf/src/encoding/encoding.ts#L201) can be used, e.g. `decodeJSON(str, AccountApplicationResponse)`. This is what I expect users will need the vast majority of the time, not the verbose code present in the request classes.

In these request classes, we are in a bit of weird situation when `prepare` function is called, since the JSON response is already parsed into an object. I have a desire to change this so that the JSON string is the input to `prepare`, then `decodeJSON` can be used and the code would be cleaner. But this PR has already grown large, so I didn't want to attempt that now.